src/js/providers/tracks-mixin.js

```diff
@@ -517,15 +517,35 @@ function addVTTCuesToTrack(track, vttCues) {
 // ////////////////////
 
 function _addCueToTrack(renderNatively, track, vttCue) {
-    if (!(Browser.ie && renderNatively) || !window.TextTrackCue) {
-        track.addCue(vttCue);
-        return;
+    let cue = vttCue;
+    if (Browser.ie && renderNatively && window.TextTrackCue) {
```

Comment:
We returned early in the non-ie use case because that code path will get hit in majority of cases (chrome, safari, ff).

The `!window.TextTrackCue` check was there to prevent this code from blowing up in environments where `window.TextTrackCue` doesn't exist and we use the vttCue polyfill.

Every browser - ie, edge, ff, safari, chrome - supports `window.TextTrackCue`, so if we were to keep this condition, you could simplify it to `if (Browser.ie && renderNatively)`.